lib/protocols/vast/KalturaVastParser.js

```diff
@@ -50,5 +57,22 @@ class KalturaVastParser
 		if(oldHeader[headerType])
 			newHeader[headerType] = oldHeader[headerType];
 	}
+
+	static _validate(vastObject)
+	{
+		if (!vastObject || vastObject.ads.length == 0)
+			return true; //will pass through the vastObject
+
+		for (let ad of vastObject.ads) {
+			for (let creative of ad.creatives) {
+				if (creative.duration <= 0 || !creative.mediaFiles || creative.mediaFiles.length == 0)
+					return false;
+				for (let mediaFile of creative.mediaFiles)
```

Comment:
I am not sure about the "of" syntax - but some of these iteration modules in node results in parsing the member functions as well , validate this is not the case for you